packages/dds/tree/src/codec/codec.ts

```diff
@@ -457,19 +452,21 @@ export enum FluidClientVersion {
 	/** Fluid Framework Client 2.52 and newer. */
 	// New formats introduced in 2.52:
 	// - DetachedFieldIndex FormatV2
-	v2_52 = 2.052,
+	v2_52: "2.52.0" as MinimumVersionForCollab,
```

Comment:
`as` casts are relatively unsafe.

I see no reasons to type erase the exact string value here by unsafely converting to MinimumVersionForCollab. Everything should work fine if you simply remove the `as MinimumVersionForCollab` and more specific typing will be preserved that way.

If you do with to have the compiler validate the values here are compatible with MinimumVersionForCollab, you can do so in any of several safer ways, like a `satisfies Record<string, MinimumVersionForCollab>` clause.